src/com/twitter/intellij/pants/settings/PantsProjectSettingsControl.java

```diff
@@ -112,65 +120,87 @@ protected boolean isExtraSettingModified() {
 
-      myTargetSpecsBox.setEmptyText(spec);
-      myTargetSpecsBox.addItem(spec, spec, true);
+        if (relativeProjectPath.isPresent()) {
+          String spec = relativeProjectPath.get() + "/::";
 
-      myLibsWithSourcesCheckBox.setEnabled(true);
-    }
-    else if (PantsUtil.isExecutable(file.getPath())) {
-      myTargetSpecsBox.setEnabled(false);
-      myTargetSpecsBox.setEmptyText(file.getName());
+          myTargetSpecsBox.setEmptyText(spec);
+          myTargetSpecsBox.addItem(spec, spec, true);
 
-      myLibsWithSourcesCheckBox.setSelected(false);
-      myLibsWithSourcesCheckBox.setEnabled(false);
-    }
-    else {
-      myTargetSpecsBox.setEnabled(true);
-      myTargetSpecsBox.setEmptyText(StatusText.DEFAULT_EMPTY_TEXT);
-      myLibsWithSourcesCheckBox.setEnabled(true);
-
-      ProgressManager.getInstance().run(new Task.Modal(getProject(), PantsBundle.message("pants.getting.target.list"), false) {
-        @Override
-        public void run(@NotNull ProgressIndicator indicator) {
-          try {
-            final Collection<String> targets = PantsUtil.listAllTargets(projectPath);
-            UIUtil.invokeLaterIfNeeded(() -> {
+          myLibsWithSourcesCheckBox.setEnabled(true);
+        } else {
+          errors.add(String.format("Fail to find relative path from %s to build root.", file.getPath()));
+          return;
+        }
+        break;
+
+      case executable:
+        myTargetSpecsBox.setEnabled(false);
+        myTargetSpecsBox.setEmptyText(PantsUtil.getRelativeProjectPath(file.getPath()).orElse(file.getName()));
+
+        myLibsWithSourcesCheckBox.setSelected(false);
+        myLibsWithSourcesCheckBox.setEnabled(false);
+        break;
+
+      case isBUILDFile:
+        myTargetSpecsBox.setEnabled(true);
+        myTargetSpecsBox.setEmptyText(StatusText.DEFAULT_EMPTY_TEXT);
+        myLibsWithSourcesCheckBox.setEnabled(true);
+
+        ProgressManager.getInstance().run(new Task.Modal(getProject(),
+            PantsBundle.message("pants.getting.target.list"), false) {
+          @Override
+          public void run(@NotNull ProgressIndicator indicator) {
+            try {
+              final Collection<String> targets = PantsUtil.listAllTargets(projectPath);
+              UIUtil.invokeLaterIfNeeded(() -> {
                 targets.forEach(s -> myTargetSpecsBox.addItem(s, s, false));
               });
-          } catch (RuntimeException e) {
-            UIUtil.invokeLaterIfNeeded((Runnable) () -> {
+            } catch (RuntimeException e) {
+              UIUtil.invokeLaterIfNeeded((Runnable) () -> {
                 Messages.showErrorDialog(getProject(), e.getMessage(), "Pants Failure");
                 Messages.createMessageDialogRemover(getProject()).run();
               });
+            }
           }
-        }
-      });
+        });
+        break;
+      default:
+        UIUtil.invokeLaterIfNeeded((Runnable) () -> {
```

Comment:
Hmm selecting a directory would trigger this..so maybe noop, which is the status quo?

Reply:
hm. I don't think so. It should hit https://github.com/pantsbuild/intellij-pants-plugin/pull/362/files/655250383b1e924edb3c14e9d144a69aa9f1c5b6#diff-42c61d4fa92b800ccf4d203aca9fc5feR151

Reply:
<img width="744" alt="screen shot 2018-07-26 at 3 10 55 pm" src="https://user-images.githubusercontent.com/596358/43291399-22f11a5c-90e6-11e8-964a-601c9672e2fd.png">

Reply:
nice